Week4/app/src/main/java/com/oliverperez/blockbusterchallenge/Adapters/MovieGridAdapter.kt

```diff
@@ -0,0 +1,35 @@
+package com.oliverperez.blockbusterchallenge.Adapters
+
+import android.media.Image
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.recyclerview.widget.RecyclerView
+import com.oliverperez.blockbusterchallenge.DataManagers.MovieDataManager
+import com.oliverperez.blockbusterchallenge.Models.Movie
+import com.oliverperez.blockbusterchallenge.R
+import com.oliverperez.blockbusterchallenge.Views.MovieViewHolder
+import kotlinx.android.synthetic.main.movie_view_holder.view.*
+
+class MovieGridAdapter(private val movies: List<Movie>, private val clickListener: MovieClickListener): RecyclerView.Adapter<MovieViewHolder>() {
```

Comment:
When passing data to adapter you want to create a separate method for it and a member variable that holds the data. Something like this: 

```kotlin
private val movies = mutableListOf<Movie>()

fun setMovies(movies: List<Movie>) {
this.movies.clear()
this.movies.addAll(movies)
notifyDatasetChanged()
}
```

Don't pass data through the constructor.